src/main/java/apoc/path/PathExplorer.java

```diff
@@ -0,0 +1,210 @@
+package apoc.path;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.neo4j.graphdb.Direction;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Path;
+import org.neo4j.graphdb.PathExpanderBuilder;
+import org.neo4j.graphdb.RelationshipType;
+import org.neo4j.graphdb.traversal.Evaluation;
+import org.neo4j.graphdb.traversal.Evaluator;
+import org.neo4j.graphdb.traversal.Evaluators;
+import org.neo4j.graphdb.traversal.TraversalDescription;
+import org.neo4j.graphdb.traversal.Uniqueness;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import apoc.Description;
+
+
+public class PathExplorer {
+	private static final String VERSION = "0.5";
+	@Context
+    public GraphDatabaseService db;
+
+	@Context
+    public Log log;
+	
+	
+	public Stream<InfoContainer> info() {
+		return getInfo().stream().map(InfoContainer::new);
+	}
+	
+	
+	
+	@Procedure("apoc.path.expand")
+	@Description("usage - CALL apoc.path.expand(startNode <id>|Node, relationshipFilter, labelFilter, minLevel, maxLevel ) yield expandedPath as <identifier>")
+	public Stream<PathContainer> explorePath(@Name("start") Object start
+			                   ,@Name("relationshipFilter") String pathFilter
+			                   ,@Name("labelFilter") String labelFilter
```

Comment:
Should label list be separated into independent b/w lists? You can get rid of string manipulation and (IMO) it's more usable that way. Learning another (however small) language might be an impediment :)

More general, can't existing Cypher syntax be used?